presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/DetermineJoinDistributionType.java

```diff
@@ -224,6 +230,13 @@ private boolean isSmall(PlanNode node)
             return !dataSize.isValueUnknown() && dataSize.getValue() < smallSizeLimit;
         }
 
+        private boolean isMuchSmaller(PlanNode node, PlanNode other)
+        {
+            Estimate nodeDataSize = getOutputSizeEstimate(node);
+            Estimate otherDataSize = getOutputSizeEstimate(other);
+            return !nodeDataSize.isValueUnknown() && !otherDataSize.isValueUnknown() && nodeDataSize.getValue() * (nodeCount - 1) < otherDataSize.getValue();
```

Comment:
I suppose in a repartitioned join you need to distribute both tables so -1 is b/c of that, e.i.
1) replicated: transfer of BUILD n times
2) repartitioned: transfer both BUILD and PROBE once
so n * BUILD < BUILD + PROBE
so (n-1) * BUILD < PROBE